### PR TITLE
fix: 691 - implemented "imagesToJson"

### PR DESCRIPTION
**Pull request by monsieurtanuki**
New file:
* `api_json_to_from_test.dart`: unit tests about to/from json conversions

Impacted files:
* `api_search_products_test.dart`: unrelated fix
* `json_helper.dart`: implemented method `imagesToJson`; refactored method `imagesFromJson`
* `product_images.dart`: added fields width and height; added `hashcode` and `==` operator

### What
- `imagesFromJson`was already implemented, but not `imagesToJson`.
- That was problematic if we wanted to store locally in a `String` a `Product` with field `images`.

### Fixes bug(s)
- Closes: #691
- Closes: #692

### CODE REVIEW

File: lib/src/model/product_image.dart [hunk 1]
```diff
@@ -179,5 +187,36 @@ class ProductImage {
       '${y1 == null ? '' : ',y1=$y1'}'
       '${x2 == null ? '' : ',x2=$x2'}'
       '${y2 == null ? '' : ',y2=$y2'}'
+      '${width == null ? '' : ',width=$width'}'
+      '${height == null ? '' : ',height=$height'}'
       ')';
+
+  @override
+  int get hashCode =>
+      '${field.offTag}_${language?.code}_${size?.offTag}'.hashCode;
```
Comment (M123-dev) [inline lib/src/model/product_image.dart:196]: Can't we just use the toString method here

**Review by M123-dev — APPROVED** [review]
Looks good, thanks @monsieurtanuki

File: lib/src/model/product_image.dart (hunk 1 above)
Comment (monsieurtanuki) [reply lib/src/model/product_image.dart:196]: In that case we could probably, but I don't think a hashcode should depend on other unrelated methods. What if the implementation of `toString` is removed from that class?

Fun facts:
1. I don't care about `hashcode`, it's just that the operator `==` triggers a warning about un-implemented `hashcode` method
2. with your logic I could have implemented `==` with `toString` too. That would have been ugly

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev for the review!

**Pull request merged**